packages/flutter/test/material/tabs_test.dart

```diff
@@ -2350,4 +2350,76 @@ void main() {
     expect(find.text('Tab1'), findsOneWidget);
     expect(find.text('Tab2'), findsOneWidget);
   });
+
+  testWidgets('DefaultTabController should allow for a length of zero', (WidgetTester tester) async {
+    List<String> tabTextContent = <String>[];
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: StatefulBuilder(
+          builder: (BuildContext context, StateSetter setState) {
+            return DefaultTabController(
+              length: tabTextContent.length,
+              child: Scaffold(
```

Comment:
This test could probably be much simpler. For example it could just rebuild the frame with different values of tabTextContent and not bother with the buttons and icons and so on. However if we're reproducing the test case that was originally used to demo the bug, it's OK.

Reply:
I think the primary motivation I had was to make sure that it not only initializes to zero tabs just fine, but also accounts for dynamically updating the number of tabs from > 0 to zero as well. I'll add a comment to clarify this.